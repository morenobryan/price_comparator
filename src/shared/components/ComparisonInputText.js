// @flow
import React from 'react';
import { Field } from 'redux-form';
import { StyleSheet, Text, TextInput, View } from 'react-native';
import type { KeyboardType, ReturnKeyType } from '../../types';
import { textColor } from '../../shared/styles';

type Props = {
  input: {
    onChange: any => void,
  },
  keyboardType: KeyboardType,
  meta: {
    touched: boolean,
    error: string,
    warning: string,
  },
  onEnter: () => void,
  refField: any,
  returnKeyType: ReturnKeyType,
  selectionColor: string,
  fieldTitle: string,
};

export default class ComparisonInput extends React.Component<Props> {
  render() {
    const {
      input: { onChange, ...restInput },
      keyboardType,
      meta: { touched, error, warning },
      onEnter,
      refField,
      returnKeyType,
      selectionColor,
      fieldTitle,
    } = this.props;

    return (
      <View>
        <View style={styles.container}>
          <Text style={styles.inputLabel}>{fieldTitle}</Text>
          <TextInput
            keyboardType={keyboardType}
            onChangeText={onChange}
            onSubmitEditing={onEnter}
            ref={refField}
            returnKeyType={returnKeyType}
            selectionColor={selectionColor}
            style={[styles.input, touched && error ? styles.errorInput : undefined]}
            underlineColorAndroid="transparent"
            {...restInput}
          />
        </View>
        <View>
          {touched &&
            ((error && <Text style={styles.error}>{error}</Text>) ||
              (warning && <Text>{warning}</Text>))}
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
  },
  inputLabel: {
    flex: 0.5,
    color: textColor,
    textAlign: 'right',
    marginRight: 10,
    fontFamily: 'proximaNovaAltRegular',
  },
  input: {
    borderBottomWidth: 1,
    borderColor: textColor,
    color: textColor,
    flex: 0.5,
    fontFamily: 'proximaNovaAltRegular',
    padding: 0,
  },
  errorInput: {
    borderColor: '#e53935',
  },
  error: {
    fontFamily: 'proximaNovaAltRegular',
    textAlign: 'right',
    color: '#e53935',
  },
});
