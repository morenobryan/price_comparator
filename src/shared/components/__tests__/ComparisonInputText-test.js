import 'react-native';
import React from 'react';
import ComparisonInputText from '../ComparisonInputText';

import renderer from 'react-test-renderer';

const props = {
  input: {
    onChange: () => {},
  },
  keyboardType: 'numeric',
  meta: {
    touched: false,
    error: 'error',
    warning: undefined,
  },
  onEnter: () => {},
  refField: 'ref',
  returnKeyType: 'next',
  selectionColor: '#fff',
  fieldTitle: 'title',
};

test('renders correctly', () => {
  const tree = renderer.create(<ComparisonInputText {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
