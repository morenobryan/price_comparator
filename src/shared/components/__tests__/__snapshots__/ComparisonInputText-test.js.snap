// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#3D3527",
          "flex": 0.5,
          "fontFamily": "proximaNovaAltRegular",
          "marginRight": 10,
          "textAlign": "right",
        }
      }
    >
      title
    </Text>
    <TextInput
      keyboardType="numeric"
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      returnKeyType="next"
      selectionColor="#fff"
      style={
        Array [
          Object {
            "borderBottomWidth": 1,
            "borderColor": "#3D3527",
            "color": "#3D3527",
            "flex": 0.5,
            "fontFamily": "proximaNovaAltRegular",
            "padding": 0,
          },
          undefined,
        ]
      }
      underlineColorAndroid="transparent"
    />
  </View>
  <View />
</View>
`;
